qemu/tests/block_resize.py

```diff
@@ -110,6 +111,15 @@ def get_md5_of_file(filename):
                                                  os_type, fstype=fstype,
                                                  labeltype=labeltype)[0]
         partition = mpoint.replace('mnt', 'dev') if 'mnt' in mpoint else None
+        # Add following code to solve the disk letter change issue on s390x
+        # virtio-blk, i.e. record the disk_uuid before guest reboot
+        if os_type == 'linux':
```

Comment:
I think this should not be a special problem on s390x, using UUID is a better choice to bind the disk.

Regarding the details of logic, you should get the UUID of the test disk before reboot, and use UUID to reverse the disk letter after reboot, instead of using the disk letter to reget the UUID, and then judge it.

```
# before reboot
disk_uuid = blkid /dev/vdx

# after reboot
disks = get_linux_disks(session)
for d in disks:
    if blkid | grep ${disk_uuid}:
        disk = d
        break
```

S I think you can discuss with x86 owner, the make the code stronger.

Reply:
@qingwangrh Can we extend to use the disk UUID to reget the disk after guest reboot? We found the disk letter would change after guest reboot.

Reply:
@PaulYuuu I removed the s390x limitation and extended the code to all linux platform. Please note as suggested by @qingwangrh, I updated to use disk serial number instead of the uuid to mark the disk, so it's easy to get the new disk letter from it after guest reboot. Thank you for the suggestion. Please help to review.

@qingwangrh Please help to review my code. Thanks for your help in finding the functions.